xscen/diagnostics.py

```diff
@@ -2,110 +2,290 @@
-                ds_tn = xr.open_zarr(
-                    sim.search(variable="tasmin").catalog.df.iloc[0]["path"]
-                )
-                ds_tx = xr.open_zarr(
-                    sim.search(variable="tasmax").catalog.df.iloc[0]["path"]
-                )
-
-                tn, tx, _ = _invert_unphysical_temperatures(
-                    ds_tn["tasmin"], ds_tx["tasmax"]
-                )
-                ds_tn["tasmin"] = tn
-                ds_tx["tasmax"] = tx
-                # TODO: History, attrs, etc.
-
-                save_to_zarr(
-                    ds=ds_tn,
-                    filename=sim.search(variable="tasmin").catalog.df.iloc[0]["path"],
-                    zarr_kwargs={"mode": "w"},
-                )
-                save_to_zarr(
-                    ds=ds_tx,
-                    filename=sim.search(variable="tasmax").catalog.df.iloc[0]["path"],
-                    zarr_kwargs={"mode": "w"},
-                )
-
-            # TODO: "clip" function for pr, sfcWind, huss/hurs, etc. < 0 (and > 1/100, when applicable).
-            #   Can we easily detect the variables that need to be clipped ? From the units ?
-
-
-def _invert_unphysical_temperatures(
-    tasmin: xr.DataArray, tasmax: xr.Dataset
-) -> tuple[xr.DataArray, xr.DataArray, xr.DataArray]:
-    """
-    Invert tasmin and tasmax points where tasmax <  tasmin.
+    ds: xr.Dataset | xr.DataArray
+        Dataset to check.
+    structure: dict
+        Dictionary with keys "dims" and "coords" containing the expected dimensions and coordinates.
+    calendar: str
+        Expected calendar. Synonyms should be detected correctly (e.g. "standard" and "gregorian").
+    start_date: str
+        To check if the dataset starts at least at this date.
+    end_date: str
+        To check if the dataset ends at least at this date.
+    variables_and_units: dict
+        Dictionary containing the expected variables and units.
+    cfchecks: dict
+        Dictionary where the key is the variable to check and the values are the cfchecks.
+        The cfchecks themselves must be a dictionary with the keys being the cfcheck names and the values being the arguments to pass to the cfcheck.
+        See `xclim.core.cfchecks` for more details.
+    freq: str
+        Expected frequency, written as the result of xr.infer_freq(ds.time).
+    missing: dict | str | list
+        String, list of strings, or dictionary where the key is the method to check for missing data and the values are the arguments to pass to the method.
+        The methods are: "missing_any", "at_least_n_valid", "missing_pct", "missing_wmo". See :py:func:`xclim.core.missing` for more details.
+    flags: dict
+        Dictionary where the key is the variable to check and the values are the flags.
+        The flags themselves must be a dictionary with the keys being the data_flags names and the values being the arguments to pass to the data_flags.
```

Comment:
```suggestion
        The flags themselves must be a dictionary with the keys being the data_flags names and the values being the arguments to pass to the data_flags.
        If `None` is passed instead of a dictionary, then xclim's default flags for the given variable are run. See :py:data:`xclim.core.utils.VARIABLES`.
```